### PR TITLE
Unreliable background priority (EXPOSUREAPP-3094)

### PR DESCRIPTION
**Pull request by kolyaopahle**
## Description
This PR removes the switch for background prioritisation from the settings fragment as they are not stable on all devices.
Im not sure if these changes are complete or if more work on ui flow is needed. Will open in draft state.

image

### CODE REVIEW

File: Corona-Warn-App/src/main/res/values-en/strings.xml [hunk 1]
```diff
@@ -633,11 +633,11 @@
     <!-- XACT: settings(background priority) - illustraction description -->
     <string name="settings_background_priority_illustration_description"/>
     <!-- XTXT: settings(background priority) - explains user what to do on card if background priority is enabled -->
-    <string name="settings_background_priority_card_body">"To disable prioritized background activity, please deactivate it in the device settings."</string>
+    <string name="settings_background_priority_card_body">"To enable / disable prioritized background activity, please change it in the device settings"</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values-en/strings.xml:636]: We don't change the english strings, that has to go over the translation process.

Comment (harambasicluka) [reply Corona-Warn-App/src/main/res/values-en/strings.xml:636]: please revert all changes in this file.

File: Corona-Warn-App/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -642,11 +642,11 @@
     <!-- XACT: settings(background priority) - illustraction description -->
     <string name="settings_background_priority_illustration_description" />
     <!-- XTXT: settings(background priority) - explains user what to do on card if background priority is enabled -->
-    <string name="settings_background_priority_card_body">"To disable prioritized background activity, please deactivate it in the device settings."</string>
+    <string name="settings_background_priority_card_body">"To enable / disable prioritized background activity, please change it in the device settings"</string>
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/values/strings.xml:645]: same as for the english file, we don't change this.

**Review by harambasicluka — DISMISSED** [review]
Only checked the strings.

File: Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml [hunk 3]
```diff
@@ -57,34 +57,84 @@
...
+                    <androidx.constraintlayout.widget.ConstraintLayout
+                        style="@style/row"
+                        android:layout_width="0dp"
+                        android:layout_height="wrap_content"
+                        android:focusable="true"
+                        app:layout_constraintEnd_toEndOf="parent"
+                        app:layout_constraintStart_toStartOf="parent"
+                        app:layout_constraintTop_toTopOf="parent">
+
+                        <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml:77]: ConstraintLayout within a ConstraintLayout should not be necessary.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/SettingsBackgroundPriorityFragment.kt [hunk 4]
```diff
@@ -40,19 +39,12 @@ class SettingsBackgroundPriorityFragment :
...
-            it.setOnClickListener {
-                val isPriorityEnabled = settingsViewModel.isBackgroundPriorityEnabled.value == true
-
-                if (!isPriorityEnabled) {
-                    (requireActivity() as MainActivity).apply {
-                        startActivitySafely(powerManagement.disableBatteryOptimizationsIntent)
-                    }
-                }
+        settingsRow.setOnClickListener {
+            (requireActivity() as MainActivity).apply {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/SettingsBackgroundPriorityFragment.kt:46]: Could (requireActivity() as MainActivity) to be null?  Add a"?." before the apply.

**Review by SabineLoss — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/SettingsBackgroundPriorityFragment.kt (hunk 4 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/SettingsBackgroundPriorityFragment.kt:46]: it cant be null but could raise an exception if no main activity is found, i could replace it with a null check that does nothing instead. Which behaviour would you prefer?

File: Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml (hunk 3 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml:77]: this is taken directly from the `include_settings_switch_row` include and is (i think) used to apply the row style to some of the children of the parent element

Comment (ralfgehrer) [reply Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml]: And now you can remove the nested ConstraintLayout, too @kolyaopahle

**Review by AlexanderAlferov — APPROVED** [review]
Tested. Looks good!

**Review by chris-cwa — APPROVED** [review]

**Pull request merged**